test/Microsoft.SqlTools.ServiceLayer.UnitTests/DisasterRecovery/BackupTests.cs

```diff
@@ -3,14 +3,62 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.DisasterRecovery;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Moq;
 using System.Threading.Tasks;
+using Xunit;
 
 namespace Microsoft.SqlTools.ServiceLayer.UnitTests.DisasterRecovery
 {
     public class BackupTests
     {
+        private TaskMetadata taskMetaData = new TaskMetadata
+        {
+            ServerName = "server name",
+            DatabaseName = "database name",
+            Name = "Backup Database", 
+            IsCancelable = true
+        };
+
+        [Fact]
+        public async Task VerifyCreateAndRunningBackupTask()
+        {
+            using (SqlTaskManager manager = new SqlTaskManager())
+            {
+                var mockUtility = new Mock<IBackupUtilities>();
+                DisasterRecoveryService service = new DisasterRecoveryService(mockUtility.Object);
+                SqlTask sqlTask = manager.CreateTask(this.taskMetaData, service.BackupTask);
+                Assert.NotNull(sqlTask);
+                Task taskToVerify = sqlTask.RunAsync().ContinueWith(Task =>
+                {
+                    Assert.Equal(sqlTask.TaskStatus, SqlTaskStatus.Succeeded);                 
+                });
+
+                await taskToVerify;
+            }
+        }
+
+        [Fact]
+        public async Task CancelBackupTask()
```

Comment:
>CancelBackupTask [](start = 26, length = 16)

it seems like there have been a couple bugs in the implementation for cancellation in this and the previous iteration.  Could you please look at this test and perhaps add additional test methods to cover those scenarios?  I would have expected tests to fail since the eventing and waiting isn't correct in the cancellation code path.